R/counts_labels.R

```diff
@@ -0,0 +1,39 @@
+#' Counts the number of observations in each category of a variable that has
+#' value labels, returns counts and percent of total.
+#' @export
+#' @title counts_labels
+#' @param data a data.frame
+#' @param x a column in 'data'. Must be a string.
+#' @import dplyr
+counts_labels <- function(data, x) {
+  if(is.factor(data[[x]])) {
+    data[[x]] <- hablar::convert_num(data[[x]])
+    cat("Factor converted to numeric.")
+  }
+  if(is.character(data[[x]])) {
+    data[[x]] <- hablar::convert_num(data[[x]])
+    cat("Factor converted to numeric.")
+  }
+  if(is.null(sjlabelled::get_labels(data[[x]]))) {
+    stop(paste0("'", x, "' does not have value labels.")) 
+  }
+  dims <- dim(data)
+  counts <- data %>%
+    select_(x) %>%
```

Comment:
Use of `select_()` and other underscored functions is deprecated. You can replace this with `dplyr::select()`

Reply:
I can get this to work by replacing `select_()` with `select()`, but when I replace `group_by_()` with `group_by()` or `count_()` with `count()` the function no longer works - I get an error "Error: Column `x` is unknown". Any ideas?

Reply:
What if you use `dplyr::group_by()` and `dplyr::count()`?

Reply:
Oh, nevermind -- I see what's going on here. It's because `x` is being passed as a character vector and not a name. Fixing this will require an understanding of tidy evaluation. Let's just leave it the way it was (with the underscore functions) for now.